src/background/utils/extension-api.ts

```diff
@@ -66,32 +67,32 @@ export async function readLocalStorage<T extends {[key: string]: any}>(defaults:
 }
 
 export async function writeSyncStorage<T extends {[key: string]: any}>(values: T): Promise<void> {
-    return new Promise<void>((resolve, reject) => {
-        isWriting = true;
+    return new Promise<void>(async (resolve, reject) => {
+        await mutexStorageWriting.lock();
         chrome.storage.sync.set(values, () => {
             if (chrome.runtime.lastError) {
                 reject(chrome.runtime.lastError);
                 return;
             }
             resolve();
-            setTimeout(() => isWriting = false);
+            setTimeout(() => mutexStorageWriting.unlock(), 500);
```

Comment:
Why 500? Is it a magic number?

Reply:
It is a magic number, but it will ensure that the promises of `saveSettingsIntoStorage` are run etc. So that the call from `chrome.storage.onChanged.addListener` can be runned. Also it should prevent that this callback is given a priority due to the none-usage of timeout Number. `500` is a pretty big number so we wouldn't hit any timing races etc.